### PR TITLE
feat!: simplify angle extension in to a half turns rotation type

### LINKED ISSUE #605: `AngleOp::adiv` is ambiguous
Angles are equivalence classes mod 2π; but division by an integer (in contrast to multiplication by an integer) is not invariant under that equivalence: e.g. 0 ≡ 2π but 0/2 ≢ (2π)/2.

I would rather like to get rid of this operation for this reason, but if we do keep it we should specify the semantics, probably by coercing the initial argument to be in [0, 2π) or [-π, π). (Note that if we chose [0, 2π) then dividing e.g. the angle -π/2 by 2 would give us 3π/4, which may be unexpected.)

### LINKED ISSUE #608: Angle should always assume maximum precision
Then a single u64 to specify the numerator is isomorphic. 
No need to track lower log_denom.
Smallest denominator rational calculation can be performed by library functions.

### LINKED ISSUE #610: Simplify angle extension to rotations
Rotation defined by number of half turns
(from/to) half turn ops

temporary addition op to support badger and pytket conversions

### PR DESCRIPTION
**Pull request by ss2165**
Closes #610
Closes #608
Closes #605

BREAKING CHANGE: "tket2.angle" extension replaced with "tket2.rotation" extension with rotation type and simplified set of operations.

### CODE REVIEW

File: tket2/src/extension/rotation.rs [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    ) -> Result<Self, hugr::extension::simple_op::OpLoadError>
+    where
+        Self: Sized,
+    {
+        hugr::extension::simple_op::try_from_name(op_def.name(), op_def.extension())
+    }
+
+    fn signature(&self) -> hugr::extension::SignatureFunc {
+        match self {
+            RotationOp::fromturns => {
```
[the remark this answers is not in the document]
Comment (ss2165) [reply tket2/src/extension/rotation.rs:134]: Yes sorry I forgot we discussed this, feel free to make the change

File: tket2/src/extension/rotation.rs [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct ConstRotation {
+    half_turns: f64,
+}
+
+impl ConstRotation {
+    /// The constant π
+    pub const PI: Self = Self::new_unchecked(1.0);
+    /// The constant 2π
+    pub const TAU: Self = Self::new_unchecked(2.0);
```
Comment (cqc-alec) [inline tket2/src/extension/rotation.rs:51]: Not sure how useful this one is. More useful might be:
```rust
    pub const ZERO: Self = Self::new_unchecked(0.0);
```

File: tket2/src/extension/rotation.rs [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+                type_row![ROTATION_TYPE],
+            ),
+        }
+        .into()
+    }
+
+    fn description(&self) -> String {
+        match self {
+            RotationOp::fromturns => {
+                "Construct rotation from number of turns (would be multiples of π in radians)."
```
Comment (cqc-alec) [inline tket2/src/extension/rotation.rs:152]:
```suggestion
                "Construct rotation from number of half turns (would be multiples of π in radians)."
```

File: tket2/src/extension/rotation.rs [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+            }
+            RotationOp::toturns => {
+                "Convert rotation to number of turns (would be multiples of π in radians)."
```
Comment (cqc-alec) [inline tket2/src/extension/rotation.rs:155]:
```suggestion
                "Convert rotation to number of half turns (would be multiples of π in radians)."
```

File: tket2/src/extension/rotation.rs [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+        }
+        .to_owned()
+    }
+
+    fn extension(&self) -> hugr::extension::ExtensionId {
+        ROTATION_EXTENSION_ID
+    }
+
+    // TODO constant folding
+    // https://github.com/CQCL/tket2/issues/405
```
Comment (cqc-alec) [inline tket2/src/extension/rotation.rs:167]: Maybe need to update this issue.

**Review by cqc-alec — COMMENTED** [review]
Just minor comments.

On a bike-shedding note, not completely happy with the name "rotation" as it suggests an operation rather than a quantity, but I can't think of a better alternative to "angle".

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**